en/docs/references/extend/federation/write-a-custom-oauth2-federated-authenticator.md

```diff
@@ -0,0 +1,84 @@
+# Write a Custom OAuth2 Federated Authenticator
+
+A custom OAuth2 authenticator can be written to authenticate a user with an external system. The external system can be
+any Identity provider supporting OAuth2 including Facebook, Kakao, Naver etc. You can use the OAuth2GenericAuthenticator
```

Comment:
IMO, it would be great if the text "OAuth2GenericAuthenticator" is a hyperlink that points to the class location in the GH repo.